qrao/magic_rounding.py

```diff
@@ -190,10 +196,10 @@ def _unpack_measurement_outcome(
             q, op = var2op[var]
             # get the index in [0,1,2] corresponding
             # to each possible Pauli.
-            op_index = self._XYZ[str(op)]
+            op_index = self._OP_INDICES[vars_per_qubit][str(op)]
```

Comment:
The comment on the previous line could probably be improved along with this change, since it previously implicitly assumed a 3-QRAC with X,Y,Z.